OPHD/StructureManager.cpp

```diff
@@ -260,19 +260,21 @@ void StructureManager::removeStructure(Structure* structure)
 	StructureList& structures = mStructureLists[structure->structureClass()];
 
 	const auto it = std::find(structures.begin(), structures.end(), structure);
-	if (it != structures.end())
+	const auto isFoundStructureTable = it != structures.end();
+	if (isFoundStructureTable)
 	{
 		structures.erase(it);
 	}
 
 	const auto tileTableIt = mStructureTileTable.find(structure);
-	if (tileTableIt != mStructureTileTable.end())
+	const auto isFoundTileTable = tileTableIt != mStructureTileTable.end();
+	if (isFoundTileTable)
 	{
 		tileTableIt->second->deleteThing();
 		mStructureTileTable.erase(tileTableIt);
 	}
 
-	if ((it == structures.end()) || (tileTableIt == mStructureTileTable.end()))
+	if (!isFoundStructureTable || !isFoundTileTable)
```

Comment:
I'm wondering if this check is even necessary anymore? I originally had the check in there to ensure that I wasn't doing stupid stuff during debugging but it seems like a relatively simple change. By removing this check the above two comparisons can be left as they were in what I would consider a more expressive format versus adding boolean flags.

If the check is deemed necessary this code looks fine, otherwise I'd remove the check and change the above two lines back to what they were.

Reply:
You're probably right. I had briefly considered just removing the check. Though I was a little concerned that with the active development we could easily re-introduce a case where we accidentally try to delete a non-existent building. I figured it might be better to leave the check intact for now.

Anyway, I'm going to merge as is, as it's the fastest way to solve the bug, though I'm quite open to removing the check in a followup PR if people want to go that route.

Reply:
Sounds good to me. I have no objections.